kijiji_manager/kijijiapi.py

```diff
@@ -381,7 +382,25 @@ def post_conversation_reply(self, user_id, token, conversation_id, ad_id, userna
             return doc
         else:
             raise KijijiApiException(self._error_reason(doc))
-
+    
+    def geo_location(self, postal_code):
+        postalcode = self.check_postal_code_length(postal_code)
```

Comment:
```suggestion
        # pgeocode.Nominatim.query_postal_code actually only needs the first three characters of the postal code to do the lookup
        postalcode = postal_code[:3]
```

`pgeocode.Nominatim.query_postal_code` actually only needs the first three characters of the postal code to do the lookup. The latter 3 postal code characters do not improve the accuracy of the result.

Reply:
Alternatively we could change the postal code validation on the form input to enforce having a space between the first 3 characters and the last 3 characters. Ensuring that there is a space means you can pass the postal code string directly to `query_postal_code` without having to modify it (it splits the two components on whitespace and uses only the first component).

Reply:
The postal code splitting was carried over from experiments with geopy (which was pickier) prior to the switch to pgeocode. Just ran a test with pgeocode and you can feed it just the first 3 characters, all 6 or six with a space. It'll give you the same results regardless. So yes, completely agree the postal code splitting is unnecessary.